### ISSUE TITLE
Cannot read property 'type' of null when linting OpenAPI files

### ISSUE THREAD

**Issue by distortedsignal** [IBM/openapi-validator#180]
I'm the same guy from [this issue](https://github.com/IBM/openapi-validator/issues/179), this is a problem with a different Open API file.

----

My team is changing over to Open API 3.0 documentation from RAML, and I'm trying to build up a linter from this tool. Our API is (fairly) complex and our data structures are deeply nested. I recently ran into an issue where one of our *.json files couldn't be linted because of a type deref on a null.

### Local information

Installed via `npm` with the `-g` flag.

In the directory that I'm linting:

```sh
> lint-openapi -v
0.28.0
```

The call that makes the error happen, and output:

```sh
> # Source filename on following line changed to protect the guilty.
> lint-openapi --debug openapi_2.json

[Error] There was a problem with a validator.
Cannot read property 'type' of null

TypeError: Cannot read property 'type' of null
    at arrayOctetSequences (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:38:20)
    at findOctetSequencePaths (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:22:34)
    at /usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:67:14
    at Array.forEach (<anonymous>)
    at objectOctetSequences (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:53:35)
    at findOctetSequencePaths (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:24:34)
    at /usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:67:14
    at Array.forEach (<anonymous>)
    at objectOctetSequences (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:53:35)
    at findOctetSequencePaths (/usr/local/lib/node_modules/ibm-openapi-validator/src/plugins/utils/findOctetSequencePaths.js:24:34)
```

I feel like this is pretty hard to debug because the file splay of our API documentation means that I'm not entirely sure where this error is being hit. Would it be possible to print the name of the file (or, even better, the name of the object/path of the object) that procs the error?

**Comment by mkistler** [comment]
At the point of this processing, I don't believe we have a filename to print.  This code is operating on a `resolvedSchema`, meaning a schema where references have already been resolved.

But there may be a clue that can help find the problem.  The code that failed looks like this:
```function arrayOctetSequences(resolvedSchema, path) {
  const arrayPathsToOctetSequence = [];
  const arrayItems = resolvedSchema.items;
  if (arrayItems !== undefined) {
    // supports both array and string (delimited by .) paths
    const pathToSchema = Array.isArray(path)
      ? path.concat('items')
      : `${path}.items`;
    if (arrayItems.type === 'string' && arrayItems.format === 'binary') {
```

The last line is the line that failed, attempting to reference `type` of `arrayItems`.  You see just above that the code checks `arrayItems !== undefined`, but does not check for `null`.  So this suggests that somewhere in your API doc there is an `items: null`.  Maybe.  Worth looking for.

**Comment by distortedsignal** [comment]
I was just about to tag somebody in the PR, but I see that there is activity on this ticket. I'll hold my peace for now.

**Comment by dpopp07** [comment]
@jorge-ibm was @mkistler going to work on this? If not, @distortedsignal, we would welcome the PR!

**Comment by jorge-ibm** [comment]
Yes I think @mkistler was going to take a look, but I agree. @distortedsignal if you have a PR for this already please feel free to submit it

**Comment by distortedsignal** [comment]
I sent a PR to add the path to the exception text. I figure this is the best way to report to the user that their OpenAPI spec is not valid is to point them to where the invalid spec is. Check out the PR here: https://github.com/IBM/openapi-validator/pull/190

**Comment by dpopp07** [comment]
Resolved in #190 

@distortedsignal Let me know if that PR doesn't fully resolve the issue. Thanks for the PR!

**Comment by distortedsignal** [comment]
I think the PR resolves the issue _enough_. It will at least tell you where the data types are that are causing the issue. This is as much as can be asked for when dealing with faulty API documentation.
